### PR TITLE
Use already cloned git repo in Dockerfile instead of git clone

### PR DESCRIPTION
**Pull request by stuggi**
To customize the image using the github action, use the already cloned repo in the job and copy the content when building the openstack-ansibleee-runner container image.

Depends-On: https://review.rdoproject.org/r/c/rdo-jobs/+/47269

### CODE REVIEW

**Comment by stuggi** [comment]
@rebtoor its related to your PR https://github.com/openstack-k8s-operators/edpm-ansible/pull/51, was just wondering why we clone the repo in the Dockerfile instead of copy in the content of the PR to get new changes coming with the PR. This would have the advantage that when you push to your fork, the github action with the repo vars set on the fork can create the image for the PR

**Comment by rebtoor** [comment]
@stuggi makes totally sense, let's wait for CI tests to finish and I'll merge this PR. Thanks a lot!

**Comment by rebtoor** [comment]
@stuggi I was able to successfully build the image with command:
`podman -t $TAG -f openstack-ansibleeee/Dockerfile .`
launched from the project root.

BTW, it seems that something should be changed on CI side during the build of the container. In the build step [here](https://review.rdoproject.org/cgit/rdo-jobs/tree/playbooks/edpm/deploy_podified_control_plane.yaml#n23) path should become the project root, the argument `file` in `build` section should be `openstack-ansibleee/Dockerfile` and probably the `EDPM_LOCAL_REPO` env var and its logic could be dropped so the `extra_vars` field isn't needed anymore.

@cjeanner @raukadah I'll left this comment here so you can change the playbook task when this PR will be merged.

File: openstack_ansibleee/Dockerfile [hunk 1]
```diff
@@ -21,13 +23,13 @@ RUN cd /var/tmp/edpm-ansible && python3 setup.py install --prefix=/usr
 # It will give Device/Resource busy. In order to fix that. Clean up the directory
 # Only when EDPM_LOCAL_REPO is defined.
 RUN if [ "$EDPM_LOCAL_REPO" == "" ] ; then \
-        echo $EDPM_LOCAL_REPO; \
+        echo EDPM_LOCAL_REPO $EDPM_LOCAL_REPO; \
         rm -fr /var/tmp/edpm-ansible; \
     fi
```
Comment (rebtoor) [inline openstack_ansibleee/Dockerfile:28]: We can drop all of this... we already use the "local" repo, we just don't need to mount it as en external volume anymore.

**Review by rebtoor — CHANGES_REQUESTED** [review]
I left a couple of comments

File: openstack_ansibleee/Dockerfile [hunk 2]
```diff
@@ -1,16 +1,18 @@
 FROM quay.io/ansible/creator-ee:v0.13.0
 ARG EDPM_LOCAL_REPO=""
```
[the remark this answers is not in the document]
Comment (stuggi) [reply openstack_ansibleee/Dockerfile:2]: ack will remove it. as initial version just wanted to do the changes for my purpose. will update

[the remark this answers is not in the document]
Comment (stuggi) [reply openstack_ansibleee/Dockerfile]: done

File: openstack_ansibleee/Dockerfile (hunk 1 above)
Comment (stuggi) [reply openstack_ansibleee/Dockerfile]: done

**Comment by raukadah** [comment]
recheck

**Review by raukadah — APPROVED** [review]
Thank you for improving it.

**Pull request merged**